server/service/osquery_utils/queries_test.go

```diff
@@ -317,17 +317,27 @@ func TestGetDetailQueries(t *testing.T) {
 }
 
 func TestDirectIngestMDM(t *testing.T) {
+	ds := new(mock.Store)
+	ds.SetOrUpdateMDMDataFunc = func(ctx context.Context, hostID uint, enrolled bool, serverURL string, installedFromDep bool) error {
+		require.False(t, enrolled)
+		require.False(t, installedFromDep)
+		require.Empty(t, serverURL)
+		return nil
+	}
+
 	var host fleet.Host
 
-	err := directIngestMDM(context.Background(), log.NewNopLogger(), &host, nil, []map[string]string{}, true)
+	err := directIngestMDM(context.Background(), log.NewNopLogger(), &host, ds, []map[string]string{}, true)
 	require.NoError(t, err)
+	require.False(t, ds.SetOrUpdateMDMDataFuncInvoked)
```

Comment:
Set `ds.SetOrUpdateMDMDataFuncInvoked=false` below

Reply:
But it's already false.

Reply:
Ya, got confused.